janitor/functions.py

```diff
@@ -4245,3 +4245,133 @@ def fill_direction(
+                           'ID' : [1,2,3,4,5,6],
+                           'result' : ["pass", "fail", "pass",
+                                       "pass", "fail", "pass"]})
+
+        # Ascending top 3:
+        df.groupby_topk('result', 'age', 3)
+        #       age  ID  result
+        #result
+        #fail   21   5   fail
+        #       22   2   fail
+        #pass   20   1   pass
+        #       22   6   pass
+        #       23   4   pass
+
+        #Descending top 2:
+        df.groupby_topk('result', 'age', 2, {'ascending':False})
+        #       age  ID result
+        #result
+        #fail   22   2   fail
+        #       21   5   fail
+        #pass   24   3   pass
+        #       23   4   pass
+
+    Functional usage syntax:
+
+    .. code-block:: python
+
+        import pandas as pd
+        import janitor as jn
+
+        df = pd.DataFrame(...)
+        df = jn.groupby_topk(
+            df = df,
+            groupby_column_name = 'groupby_column',
+            sort_column_name = 'sort_column',
+            k = 5
+            )
+
+    Method-chaining usage syntax:
+
+    .. code-block:: python
+
+        import pandas as pd
+        import janitor as jn
+
+        df = (
+            pd.DataFrame(...)
+            .groupby_topk(
+            df = df,
+            groupby_column_name = 'groupby_column',
+            sort_column_name = 'sort_column',
+            k = 5
+            )
+        )
+
+    :param df: A pandas dataframe.
+    :param groupby_column_name: Column name to group input dataframe `df` by.
+    :param sort_column_name: Name of the column to sort along the
+        input dataframe `df`.
+    :param k: Number of top rows to return from each group after sorting.
+    :param sort_values_kwargs: Arguments to be passed to sort_values function.
+    :returns: A pandas dataframe with top `k` rows that are grouped by
+        `groupby_column_name` column with each group sorted along the
+        column `sort_column_name`.
+    :raises: ValueError if `k` is less than 1.
+    :raises: ValueError if `groupb y_column_name` does not exist in dataframe.
+    :raises: ValueError if `sort_column_name` does not exist in dataframe.
+    :raises: ValueError if `groupby_column_name` is same as `sort_column_name`.
+    :raises: KeyError if `inplace:True` is present in `sort_values_kwargs`.
+    """
+
+    # Convert the default sort_values_kwargs from None to empty Dict
+    sort_values_kwargs = sort_values_kwargs or {}
+
+    # Check if k is greater than 0.
+    if k < 1:
+        raise ValueError(
+            "Numbers of rows per group to be returned must be greater than 0."
+        )
+
+    # Check if groupby_column_name exists in dataframe.
+    if groupby_column_name not in df.columns:
+        raise ValueError(
+            f"Column {groupby_column_name} does not exist in the dataframe."
+        )
+
+    # Check if sort_column_name exists in dataframe.
+    if sort_column_name not in df.columns:
```

Comment:
I suggest removing this if condition - Pandas would trigger the error anyway, so let Pandas worry about that and include a test for it. I believe it will reduce the number of lines of code, and maintenance as well. (This is prompting me to revisit some of my own code and refactor!). In the test, it should be KeyError, as @ericmjl suggested.

Reply:
I've done as you suggested.